feat: add tests to cover add exchange usecase

flaminic opened the pull request.

tomzemp commented on src/pages/editItem.test.js:
```diff
@@ -0,0 +1,266 @@
+
+    it('should display a warning if the user does not have permissions to add an exchange', async () => {
+        const { screen } = setUp(<EditItem />, {
+            userContext: noPermissionsUserContext(),
```
Also I think that the app is probably currently wrong in that I don't think it's checking if you have edit privileges on the current exchange, but assumes that you do if you have view access 🫤. I guess we should have a read only mode in that case.

tomzemp reviewed (approved):
Looks good!

I added one more overaching comment on the setup for the editItem page.

There's a lot of additional tests that we could do here as well. Not sure if it would help defining some as part of this work even if we don't want to implement them at the moment? Some ideas that come to mind:

addItem
- save does not execute and validation error appears if name is missing
- save does not execute and validation error appears if authentication details are missing (external exchange)
- gives modal warning about unsaved changes if you click cancel after making changes in the form [+ gives no warning modal if no unsaved changes, abandons changes if you click cancel from modal]
- gives validation warnings when creating a request
- gives warning modal about unsaved changes if click cancel after editing request form
- parses the ID schemes for attributes correctly when saved (i.e. you end up with ATTRIBUTE:{attribute_id} for the id scheme)
- parses UI inputs to exchange format on save (e.g. test that given certain UI inputs, the useDataMutation is called with expected exchange format) 

editItem
- parses and correctly displays ID Scheme choices (ID/Code/Attribute)
- for external exchanges, there is a button that must be clicked to edit target information / ID schemes
- for external exchanges, clicking on edit target information / edit ID schemes enables edit of both target information / ID schemes
- for external exchanges, validation warnings appear and prevent save from being called when ID schemes are edited

flaminic commented:
> for external exchanges, there is a button that must be clicked to edit target information / ID schemes



> Looks good!
> 
> I added one more overaching comment on the setup for the editItem page.
> 
> There's a lot of additional tests that we could do here as well. Not sure if it would help defining some as part of this work even if we don't want to implement them at the moment? Some ideas that come to mind:
> 
> addItem
> 
>     [x] save does not execute and validation error appears if name is missing
> 
>     [x] save does not execute and validation error appears if authentication details are missing (external exchange)
> 
>     [x] gives modal warning about unsaved changes if you click cancel after making changes in the form [+ gives no warning modal if no unsaved changes, abandons changes if you click cancel from modal]
> 
>     [x] gives validation warnings when creating a request
> 
>     [x] gives warning modal about unsaved changes if click cancel after editing request form
> 
>     [] parses the ID schemes for attributes correctly when saved (i.e. you end up with ATTRIBUTE:{attribute_id} for the id scheme)
> 
>     [] parses UI inputs to exchange format on save (e.g. test that given certain UI inputs, the useDataMutation is called with expected exchange format)
> 
> 
> editItem
> 
>     [] parses and correctly displays ID Scheme choices (ID/Code/Attribute)
> 
>     [x] for external exchanges, there is a button that must be clicked to edit target information / ID schemes
> 
>     [x] for external exchanges, clicking on edit target information / edit ID schemes enables edit of both target information / ID schemes
> 
>     * for external exchanges, validation warnings appear and prevent save from being called when ID schemes are edited

Hey @tomzemp , i've added the tests for some of the above, let's talk about the others soon and see if maybe we can test those together?
